server/src/main/java/org/opensearch/tasks/ProtobufTask.java

```diff
@@ -0,0 +1,456 @@
+/*
+* SPDX-License-Identifier: Apache-2.0
+*
+* The OpenSearch Contributors require contributions made to
+* this file be licensed under the Apache-2.0 license or a
+* compatible open source license.
+*
+*/
+
+package org.opensearch.tasks;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.ExceptionsHelper;
+import org.opensearch.action.ProtobufActionResponse;
+import org.opensearch.action.NotifyOnceListener;
+import org.opensearch.cluster.node.DiscoveryNode;
+import org.opensearch.core.common.io.stream.NamedWriteable;
+import org.opensearch.core.xcontent.ToXContent;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.tasks.proto.TaskResourceStatsProto;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Current protobuf task information
+*
+* @opensearch.internal
+*/
+public class ProtobufTask {
```

Comment:
This class is nearly identical to the existing Task, why do we need a largely identical protobuf version? This seems to imply there is coupling between the existing OpenSearch data models and their serialized form.  Can we decompose these relationships more?